### DISCUSSION TITLE
error: "int" is not defined (reportUndefinedVariable)

### DISCUSSION THREAD

**Discussion by d-tw** [microsoft/pyright#6058] [Q&A]
Hi,

I'm encountering an odd issue where pyright errors around builtin types, for example:

```
error: "int" is not defined (reportUndefinedVariable)
```

I'm assuming this is not a bug in pyright, but instead due to the way that pyright is being invoked. We're using it with [Pants](https://www.pantsbuild.org/docs/welcome-to-pants), which is a hermetic build system, similar to Bazel.

Each sub-command invoked by pants occurs within a pseudo-chroot, with only the subset of files/data required to actually execute the command in a deterministic way. I'm assuming that some file/data is not being made available to pyright, causing this error.

Can anyone shed any light on the circumstances that would occur for pyright to fail like this?

**Comment by erictraut** [comment]
Pyright includes a copy of the typeshed stdlib stubs. These stubs define type information for the `builtins` module. Based on the symptoms you're describing, it sounds like pyright doesn't have access to these typeshed stubs.
